### ISSUE TITLE
Getting Stateful Set error while deploying elasticsearch in EKS cluster

### ISSUE THREAD

**Issue by Dhanabalan-pop** [elastic/helm-charts#1241]
**Chart version:**
Latest
**Kubernetes version:**
1.19
**Kubernetes provider:** E.g. GKE (Google Kubernetes Engine)
EKS
**Helm Version:**
v3

**`helm get release` output**

e.g. `helm get elasticsearch` (replace `elasticsearch` with the name of your helm release)

*Be careful to obfuscate every secrets (credentials, token, public IP, ...) that could be visible in the output before copy-pasting.*

*If you find some secrets in plain text in `helm get release` output you should use [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to managed them is a secure way (see [Security Example](https://github.com/elastic/helm-charts/blob/master/elasticsearch/examples/security/values.yaml#L23-L38)).*

<details>
<summary>Output of helm get release</summary>

```

Error: template: elasticsearch/templates/statefulset.yaml:298:27: executing "elasticsearch/templates/statefulset.yaml" at <.Values.roles.master>: can't evaluate field master in type interface {}
```

</details>

**Describe the bug:**

**Steps to reproduce:**

1.
2.
3.

**Expected behavior:**

**Provide logs and/or server output (if relevant):**

*Be careful to obfuscate every secrets (credentials, token, public IP, ...) that could be visible in the output before copy-pasting*

```
```

**Any additional context:**

**Comment by Dhanabalan-pop** [comment]
when I try to install helm install elasticsearch using below commands, receiving error
helm repo add elastic https://helm.elastic.co
curl -O https://raw.githubusercontent.com/elastic/helm-charts/master/elasticsearch/values.yaml
helm install --name elasticsearch elastic/elasticsearch -f ./values.yaml

Error:

Error: template: elasticsearch/templates/statefulset.yaml:298:27: executing "elasticsearch/templates/statefulset.yaml" at <.Values.roles.master>: can't evaluate field master in type interface {}

**Comment by knyuwork** [comment]
I am also facing this issue but I think this has been fixed in 8.0.0-snapshot?
https://github.com/elastic/helm-charts/commit/0626843a31bacc7954561b26cca7b413e951cb9f

So the only question is, when will the 8.0.0 chart be published.

**Comment by jmlrt** [comment]
> I am also facing this issue but I think this is due to the breaking change in 8.0.0-snapshot?
> [0626843](https://github.com/elastic/helm-charts/commit/0626843a31bacc7954561b26cca7b413e951cb9f)
> 
> So the only question is, when will the 8.0.0 chart be published.

Hi @knyuwork, 8.0.0-SNAPSHOT is for development only tracking the https://github.com/elastic/elasticsearch master branch for futur 8.0.0 release.

8.0.0 chart will be released only when Elasticsearch 8.0.0 will be released.

**Comment by jmlrt** [comment]
>curl -O https://raw.githubusercontent.com/elastic/helm-charts/master/elasticsearch/values.yaml
>helm install --name elasticsearch elastic/elasticsearch -f ./values.yaml

Elasticsearch values from master branches are only compatible with the future 8.0.0 which isn't released yet as mentioned in the previous comment.

When you are running `helm install elasticsearch elastic/elasticsearch` command, you are deploying the last released version (currently [7.13.2](https://github.com/elastic/helm-charts/releases/tag/v7.13.2)).

Note that with Helm, when you're deploying the chart all the default values from the `values.yaml` file are already deployed without adding `-f ./values.yaml`. You should use create your own values file and use `-f ./values.yaml` only to override some specific values.

**Comment by botzill** [comment]
OK, but how we install version 8.0.0 for tests only?
